apisix/router.lua

```diff
@@ -22,34 +22,147 @@ local plugin_checker = require("apisix.plugin").plugin_checker
 local str_lower = string.lower
 local error   = error
 local ipairs  = ipairs
-
+local sub_str      = string.sub
+local table        = require("apisix.core.table")
+local json         = require("apisix.core.json")
 
 local _M = {version = 0.3}
 
+local function short_key(self, str)
+    return sub_str(str, #self.key + 2)
+end
 
-local function filter(route)
+local function filter(route, pre_route_obj, size)
```

Comment:
The function prototype is too obscure, `pre_route_obj` may contain different types. It's better to split the radix tree reconstruction into another function, e.g. `filter_all_routes`.